ktor-network/jvm/src/io/ktor/network/sockets/DatagramSendChannel.kt

```diff
@@ -146,7 +188,9 @@ private fun failInvokeOnClose(handler: ((cause: Throwable?) -> Unit)?) {
     throw IllegalStateException(message)
 }
 
-private fun Datagram.writeMessageTo(buffer: ByteBuffer) {
-    packet.readAvailable(buffer)
```

Comment:
readAvailable is limited to 8192 bytes, which is now fixed. A `readTo` function for ByteBuffer is missing, so implemented an alternative solution instead.

Reply:
Found the `readFully` function which does what is needed. (readFully also only read up to 8192 bytes which is fixed now)